### PR TITLE
Add `Licensed` To Help Verify Prod Licenses

### PR DESCRIPTION
**Pull request by thboop**
GitHub has a tool called [licensed](https://github.com/github/licensed) which helps us to verify that the node modules we are using are appropriately licensed for what we are using them for. It also helps to verify that the license a node module claims to be under matches the license it.

If you were previously checking in a license in the dist file, this can replace that flow.

This PR adds:
- A workflow to check licenses on pull requests and pushes to the `main` branch
- A `licensed.yml` file used to configure `licensed`
- A number of files into the `.licenses` directory which contain our dependencies and their appropriate licenses

## How does this impact me?
- You may need to [locally install](https://github.com/github/licensed#installation) licensed and run `licensed cache` to update the dependency cache if you install a new production dependency. 
  - If `licensed cache` is unable to determine the dependency, you may need to modify the cache file yourself to put the correct license. 
- You should still verify the dependency, licensed in a tool to help, but is not a substitute for [human review of dependencies](https://github.com/github/licensed#disclaimer)
- Currently, this PR only targets production dependencies, dev dependencies are not included.

### CODE REVIEW

File: .github/workflows/licensed.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    name: Check licenses
+    steps:
+      - uses: actions/checkout@v2
+      - run: npm ci
+      - name: Install licensed
+        run: |
+          cd $RUNNER_TEMP
+          curl -Lfs -o licensed.tar.gz https://github.com/github/licensed/releases/download/2.12.2/licensed-2.12.2-linux-x64.tar.gz
+          sudo tar -xzf licensed.tar.gz
+          sudo mv licensed /usr/local/bin/licensed
```
Comment (yacaovsnc) [inline .github/workflows/licensed.yml:19]: Is there no `latest` link that we can use?  Or are we pinned to this version intentionally?

**Review by yacaovsnc — APPROVED** [review]

File: .github/workflows/licensed.yml (hunk 1 above)
Comment (yacaovsnc) [reply .github/workflows/licensed.yml:19]: https://docs.github.com/en/github/administering-a-repository/linking-to-releases

**Comment by yacaovsnc** [comment]
Should we also update the contribution md file to add this instruction?

File: .licensed.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+sources:
```
Comment (yacaovsnc) [inline .licensed.yml:1]: How is this file generated and used? Do we have a list of all acceptable licenses?

For example, I saw a `- 0bsd` license in one of the sibling PRs and it's not here.  If I am contributing, I maybe a bit confused for my next step.  I wouldn't know if I should add the license here manually or it means that the lib is not acceptable in this project?

File: .github/workflows/licensed.yml (hunk 1 above)
Comment (thboop) [reply .github/workflows/licensed.yml:19]: the problem is the asset name has the version in the zip. We shouldn't really need to upgrade this that often, and if we do, its probably due to the workflow failing so we will get visibility into needing an upgrade.

**Comment by thboop** [comment]
> Should we also update the contribution md file to add this instruction?
Done!

**Pull request merged**